File: tests/transpiler/x/java/update_stmt.java
public class Main {
    static class Person {
        String name;
        int age;
        String status;
        Person(String name, int age, String status) {
            this.name = name;
            this.age = age;
            this.status = status;
        }
        @Override public String toString() {
            return String.format("{'name': %s, 'age': %s, 'status': %s}", name, age, status);
        }
    }

    static Person[] people = new Person[]{new Person("Alice", 17, "minor"), new Person("Bob", 25, "unknown"), new Person("Charlie", 18, "unknown"), new Person("Diana", 16, "minor")};

    public static void main(String[] args) {
        for (int i = 0; i < people.length; i++) {
            var item = people[i];
            if (item.age >= 18) {
                item.status = "adult";
                item.age = item.age + 1;
            }
            people[i] = item;
        }
        System.out.println("ok");
    }
}

File: tests/transpiler/x/fs/update_stmt.fs
// Generated 2025-07-21 21:44 +0700

type Person = {
    name: string
    age: int
    status: string
}
let people: struct list = [{ name = "Alice"; age = 17; status = "minor" }; { name = "Bob"; age = 25; status = "unknown" }; { name = "Charlie"; age = 18; status = "unknown" }; { name = "Diana"; age = 16; status = "minor" }]
people <- List.map (fun item -> if age >= 18 then { { item with status = "adult" } with age = age + 1 } else item) people
printfn "%s" "ok"
